codeview/src/main/java/io/github/kbiakov/codeview/Highlighter.kt

```diff
@@ -0,0 +1,141 @@
+package io.github.kbiakov.codeview
+
+import android.content.Context
+import io.github.kbiakov.codeview.highlight.ColorTheme
+import io.github.kbiakov.codeview.highlight.ColorThemeData
+
+data class Highlighter(val context: Context) {
```

Comment:
You are used Kotlin ```data class```. It's not necessary in this case, because you have only one ```data```-param (```context```) and your class contains a lot of code inside: this is not idiomatic. As marked above, refactor your code & implement Builder pattern.